docs/source/installation/for_dev.rst

```diff
@@ -0,0 +1,128 @@
+
+
+If you are a Windows developer and want to use PowerShell, you can use the below command.
+
+.. code-block:: powershell
+	
+	(Invoke-WebRequest -Uri https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py -UseBasicParsing).Content | python
+
+It will add it to path automatically. In order to make sure you have poetry installed correctly, try running:
+
+.. code-block:: bash
+
+	poetry --version
+
+
+.. note:: You may need to restart your shell in order for the changes to take effect.
+
+See the `docs on installation python poetry for more information
+<https://python-poetry.org/docs/>`_.
+
+Installing System Dependencies
+******************************
+
+Please follow the instructions under :ref:`installing-manim` to install all dependencies. Then come back to this page to continue with the installation.
+
+.. important:: Windows Users can Ignore Pycairo Installation.
+
+
+Additionally, ``git`` has to be installed. For instructions see `Documentation
+<https://git-scm.com/>`_.
+
+.. _Install Manim Poetry:
+Installing Manim using Poetry
+*****************************
+
+1.  First, clone the Manim repo locally using git.
+
+    .. code-block:: bash
+		
+		git clone https://github.com/ManimCommunity/manim.git
+
+    or
+
+    .. code-block:: bash
+		
+		git clone git@github.com:ManimCommunity/manim.git
+
+2.  Open a Terminal/Powershell/Command Prompt and cd into the cloned directory.
+
+    .. code-block:: bash
+		
+		cd path/to/manim
+    
+
+    .. note:: This path should contain a file called `pyproject.toml` if it doesn't contain it, you would need to go a level up.
+
+3.  Use the below command to install python dependencies. This will use the default python version installed.
+
+    .. code-block:: bash
+	
+         poetry install
+
+
+    .. note:: Poetry will create a virtual environment rooted at the current directory.
+    You can select the desired python version using 
+
+    .. code-block:: bash
+	
+         poetry env use <python version you need>
+
+    For example you can use for python 3.7.
+
+    .. code-block:: bash
+	
+         poetry env use 3.7
+     
+    For more information about this you can visit the `docs
+    <https://python-poetry.org/docs/managing-environments/>`_.
+
+4. Now you can activate the virtual environment ``Poetry`` has created by using the command below.
+
+  .. code-block:: bash
+
+       poetry shell
+    
+  Or if you want to run a single command use
+
+  .. code-block:: bash
+
+       poetry run manim -h
+
+Now you are free to start developing on the Manim Repository.
+
+Running the Tests Using Poetry
+******************************
+
+Once you are done with :ref:`Install Manim Poetry`, you can run manim's test suite by activating a shell using ``poetry shell`` command and then running the command ``pytest`` to run the tests. 
+
+.. important:: You should always run the test suite before making a PR. For other contributing guidelines, see :ref:`Contributing`
```

Comment:
Unrecognized reference; this seems to be a tricky one. Changing it to ``:doc:`contributing``` does not seem to solve it, possibly because it comes from markdown file.

This is something that might have to be solved in a follow-up issue.